src/main/java/com/minecolonies/coremod/entity/ai/citizen/planter/EntityAIWorkPlanter.java

```diff
@@ -164,38 +206,51 @@ protected IAIState decide()
 
         final BuildingPlantation plantation = getOwnBuilding();
 
-        final List<BlockPos> list = plantation.getPosForPhase();
-        for (final BlockPos pos : list)
+        final Item currentItem = plantation.getNextPhase();
+        final List<Item> availablePlants = plantation.getAvailablePlants();
+        final List<BuildingPlantation.PlantationItemPosition> list = plantation.getAllWorkPositions();
+
+        // Iterate all positions to check if there's a valid block or if there's anything to harvest
+        for (final BuildingPlantation.PlantationItemPosition plantationItemPosition : list)
```

Comment:
while we're on reworking a bit, I know this is additional work but it probably would be nice if the work would actively walk and "inspect" the fields. This gives a much more lively/active impression to the players watching. Would probably be another new AI State "INSPECT" or so

Reply:
This would probably be the lowest priority action right? So that state would occur all the way at the end instead of `START_WORKING` state?
Also it'd probably be best if this selects a random location of the allowed plants right?
How would I prevent him constantly walking around but giving it some delay?

Also how would this work alongside the new leisure system and inactive workers?

Reply:
I think you could have a % chance to go into the INSPECT state, which would solve the doing it constantly part. 
The locations could be in-order if we sort the list beforehand. Maybe sorting by blockpos and then having the "walk to pos" have a bigger distance, like 5 blocks so he  seems to check out patches/areas at once. Probably want to save a current list index then.

Reply:
I'd do it two-fold, you move to the blockpos in question with a bigger minimum distance of e.g. 5 to "look" if something needs to harvested or w/e, and then if you see sth that needs to then you go closer and do Actions at the block.